classes/sample_set.php

```diff
@@ -30,18 +30,53 @@ class sample_set {
     public $samples = [];
     public $starttime;
 
+    public $limit;
 
-    public function __construct($name, $starttime) {
+    public $filterrate = 1;
+    public $current = 0;
+
+    /**
+     * Constructs the sample set.
+     *
+     * @param string $name
+     * @param float $starttime
+     * @param float|null $limit
+     */
+    public function __construct(string $name, float $starttime, int $limit = null) {
```

Comment:
Curious when `$limit` might be null. I haven't read through the rest yet at the time of writing this, but seems like some assumptions made that it is set.

Reply:
It was my original idea to be able to switch it off by setting it to null, but now it has been decided to always set it to a value.